src/main/java/com/qunite/api/service/QueueServiceImpl.java

```diff
@@ -48,12 +59,21 @@ public Queue update(Queue queue, String username) {
 
   @Override
   @Transactional
-  public void enrollMember(String username, Long queueId) {
+  public Optional<Integer> enrollMember(String username, Long queueId) {
+    var user = self.saveEntry(username, queueId);
+    return self.getMemberPosition(user.getId(), queueId);
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public User saveEntry(String username, Long queueId) {
     var queue = findById(queueId).orElseThrow(
         () -> new QueueNotFoundException("Could not find queue by id %d".formatted(queueId)));
-    userRepository.findByUsername(username)
-        .map(user -> new Entry(user, queue))
-        .ifPresent(entry -> entry.getQueue().addEntry(entry));
+    var user = userRepository.findByUsername(username).orElseThrow(
+        () -> new UserNotFoundException("Could not find user by username %s".formatted(username)));
+    if (!entryRepository.existsById(new EntryId(user.getId(), queueId))) {
```

Comment:
Imho - this shouldn`t be idempotent, it`s better to throw an exception and inform that user is already in queue, it would be easier for the fe and if so, replace put with post.